### ISSUE TITLE
Initial sync is always returning 500 for one of my storages

### ISSUE THREAD

**Issue by telotortium** [pimutils/vdirsyncer#992]
I have recently been unable to complete a full sync with the CalDAV server at work. The server seems to time out after around 30 seconds on the initial REPORT, returning a HTTP 500.

* Your vdirsyncer version: vdirsyncer/0.18.1.dev134+g5870a99.d20220524 (local build, but based on recent master)
* If applicable, which server software (and which version) you're using: internal corporate CalDAV server
* Your Python version: 3.8.9
* Your operating system: Version 12.4
* Your config file:

```
[storage calendar_ro]
type = "caldav"
url = "https://caldav.example.com"
username = "john.doe"
password = "***"
read_only = true
item_types = ["VEVENT"]
```

* Use `vdirsyncer -vdebug` for debug output. The output is sensitive, but
  please attach at least the last few lines before the error (if applicable),
  censored as necessary. This is almost always the most useful information.

```
debug: ====================
debug: REPORT https://caldav.larksuite.com/john.doe/CALID/
debug: {'User-Agent': 'vdirsyncer/0.18.1.dev134+g5870a99.d20220524', 'Content-Type': 'application/xml; charset=UTF-8', 'Depth': '1'}
debug: b'<?xml version="1.0" encoding="utf-8" ?>\n            <C:calendar-query xmlns="DAV:"\n                xmlns:C="urn:ietf:params:xml:ns:caldav">\n                <prop>\n                    <getcontenttype/>\n                    <getetag/>\n                </prop>\n                <C:filter>\n                \n                <C:comp-filter name="VCALENDAR">\n                    <C:comp-filter name="VEVENT">\n                        \n                    </C:comp-filter>\n                </C:comp-filter>\n                \n                </C:filter>\n            </C:calendar-query>'
debug: Sending request...
debug: 500
debug: <CIMultiDictProxy('Server': 'AkamaiGHost', 'Mime-Version': '1.0', 'Content-Type': 'text/html', 'Content-Length': '285', 'Expires': 'Tue, 28 Jun 2022 20:31:08 GMT', 'Date': 'Tue, 28 Jun 2022 20:31:08 GMT', 'X-Cache': 'TCP_MISS from a23-195-46-36.deploy.akamaitechnologies.com (AkamaiGHost/10.8.2-41841244) (-)', 'Connection': 'keep-alive', 'x-tt-trace-tag': 'id=16;cdn-cache=miss;type=dyn', 'Server-Timing': 'cdn-cache; desc=MISS, edge; dur=30683, origin; dur=0', 'X-Akamai-Request-ID': '50ec13d2')>
error: Unknown error occurred for my_pair/my_collection: 500, message='Internal Server Error', url=URL('https://caldav.larksuite.com/john.doe/CALID/')
error: Use `-vdebug` to see the full traceback.
debug:   File "/Users/bytedance/misc/build/vdirsyncer/vdirsyncer/cli/tasks.py", line 72, in sync_collection
debug:     await sync.sync(
debug:   File "/Users/bytedance/misc/build/vdirsyncer/vdirsyncer/sync/__init__.py", line 144, in sync
debug:     a_nonempty = await a_info.prepare_new_status()
debug:   File "/Users/bytedance/misc/build/vdirsyncer/vdirsyncer/sync/__init__.py", line 48, in prepare_new_status
debug:     async for href, etag in self.storage.list():
debug:   File "/Users/bytedance/misc/build/vdirsyncer/vdirsyncer/storage/dav.py", line 899, in list
debug:     response = await self.session.request(
debug:   File "/Users/bytedance/misc/build/vdirsyncer/vdirsyncer/storage/dav.py", line 411, in request
debug:     return await http.request(method, url, session=session, **more)
debug:   File "/Users/bytedance/misc/build/vdirsyncer/vdirsyncer/http.py", line 169, in request
debug:     r.raise_for_status()
debug:   File "/Users/bytedance/.local/pipx/venvs/vdirsyncer/lib/python3.8/site-packages/aiohttp/client_reqrep.py", line 1004, in raise_for_status
debug:     raise ClientResponseError(
```

**Comment by telotortium** [comment]
I used mitmproxy to intercept the CalDAV requests from the macOS calendar application. It seems that it sets a filter on the initial REPORT to only download events close to today. Since that works fine for my purposes, I've decided to try to do this with vdirsyncer as well. Fortunately, I found the `start_date` and `end_date` options in the [config](http://vdirsyncer.pimutils.org/en/stable/config.html#storage-caldav), which work well for me:

```
start_date = "datetime.now() - timedelta(days=14)"
end_date = "datetime.now() + timedelta(days=28)"
```

Setting the filter allows the sync to complete.
